topic/machine-learning/classification-automl/automl_classification_with_pycaret.ipynb

```diff
@@ -0,0 +1,3441 @@
+    "  No, or No internet service).\n",
+    "- `StreamingTV`: Indicates if the customer has streaming TV service (Yes, No, \n",
+    "  or No internet service).\n",
+    "- `StreamingMovies`: Indicates if the customer has streaming movies service \n",
+    "  (Yes, No, or No internet service).\n",
+    "- `Contract`: The type of contract the customer has (Month-to-month, One year, \n",
+    "  Two years).\n",
+    "- `PaperlessBilling`: Indicates whether the customer has paperless billing (Yes) \n",
+    "  or not (No).\n",
+    "- `PaymentMethod`: The customer's payment method (Electronic check, \n",
+    "  Mailed check, Credit card (automatic), etc.).\n",
+    "- `MonthlyCharges`: The amount charged to the customer each month.\n",
+    "- `TotalCharges`: The total amount charged to the customer over the time they \n",
+    "  have been with the company.\n",
+    "- `Churn`: Indicates whether the customer has churned (Yes) or not (No).\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Getting started\n",
+    "\n",
+    "First, install the required dependencies. \n",
+    "\n",
+    "```bash\n",
+    "pip install -r requirements.txt\n",
+    "```\n",
+    "\n",
+    "> **Note:** As of time of this writing, Pycaret requires Python 3.8, 3.9 or \n",
+    "> 3.10.\n",
+    "\n",
+    "Secondly, we need a CrateDB instance to store and serve the data. The easiest\n",
+    "way to get started is to use the free CrateDB cloud offering. You can sign up\n",
+    "for a free account at https://console.crate.io.\n",
+    "\n",
+    "[Deploy a cluster](https://cratedb.com/docs/cloud/en/latest/tutorials/deploy/stripe.html#deploy-cluster)\n",
+    "and we're ready to go.\n",
+    "\n",
+    "Finally, create a file `.env` with the following content:\n",
+    "\n",
+    "```env\n",
+    "CRATE_HOST=<your-crate-host> # set this to localhost if you're running crate locally\n",
+    "CRATE_USER=<your-crate-user> # set this to crate if you're running crate locally\n",
+    "CRATE_PASSWORD=<your-crate-password> # set this to \"\" if you're running crate locally\n",
+    "CRATE_SSL=true # set this to false if you're running crate locally\n",
+    "```\n",
+    "\n",
+    "You can find your cratedb credentials in your \n",
+    "[CrateDB Cloud Console](https://cratedb.com/docs/cloud/en/latest/reference/overview.html#cluster).\n",
+    "\n",
+    "### Creating demo data\n",
+    "\n",
+    "For convenience, this notebook comes with an accompanying csv dataset which we \n",
+    "can quickly import in our database. Upload the csv file to your CrateDB cloud\n",
+    "cluster, as described [here](https://cratedb.com/docs/cloud/en/latest/reference/overview.html#import).\n",
+    "To follow this notebook, choose `pycaret_churn` for your database name.\n",
+    "\n",
+    "This will automatically create a new crate table and import the data.\n"
+   ]
+  },
+  {
```

Comment:
We should add the import of data in the notebook as well for the .env configuration


```suggestion
{
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If you prefer to use code to import your data, please execute the following lines which read the CSV file into a pandas data frame, automatically creates the table in CrateDB and loads the data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
   ],
   "source": [
    "import os\n",
    "import dotenv\n",
    "import sqlalchemy as sa\n",
    "import pandas as pd\n",
    "\n",
    "if os.path.exists(\".env\"):\n",
    "    dotenv.load_dotenv(\".env\", override=True)\n",
    "\n",
    "dburi = f\"crate://{os.environ['CRATE_USER']}:{os.environ['CRATE_PASSWORD']}@{os.environ['CRATE_HOST']}:4200?ssl={os.environ['CRATE_SSL']}\"\n",
    "engine = sa.create_engine(dburi, echo=True)\n",
    "\n",
    "df = pd.read_csv(\"churn-dataset.csv\")\n",
    "with engine.connect() as conn:\n",
    "    df.to_sql(\"pycaret_churn\", conn, index=False, chunksize=1000, if_exists=\"replace\")"
   ]
  },
```

Reply:
Good idea. I will add a corresponding helper routine to `cratedb-toolkit`, and will let you know when it's available by submitting a patch here.